### PR TITLE
feat(tekton): create paginated table using Backstage core components

### PR DESCRIPTION
**Pull request by debsmita1**
**Resolves:**
https://github.com/janus-idp/backstage-plugins/issues/488

**GIF:**

https://github.com/janus-idp/backstage-plugins/assets/22490998/f6057f72-4e65-4581-ba63-acc879225c83

### CODE REVIEW

File: plugins/tekton/src/components/PipelineRunList/PipelineRunHeader.tsx [hunk 1]
```diff
@@ -1,40 +1,75 @@
...
+import { chart_color_green_400 as tektonGroupColor } from '@patternfly/react-tokens/dist/js/chart_color_green_400';
 
-export default PipelineRunHeader;
+import { PipelineRunKind } from '../../types/pipelineRun';
+import { pipelineRunDuration } from '../../utils/tekton-utils';
+import LinkedPipelineRunTaskStatus from './LinkedPipelineRunTaskStatus';
+import PlrStatus from './PlrStatus';
+import ResourceBadge from './ResourceBadge';
+
+import './PipelineRunHeader.css';
```
Comment (invincibleJai) [inline plugins/tekton/src/components/PipelineRunList/PipelineRunHeader.tsx:14]: nit: keep styles import at last

File: plugins/tekton/src/components/PipelineRunList/LinkedPipelineRunTaskStatus.tsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ const LinkedPipelineRunTaskStatus = ({
   pipelineRun,
 }: LinkedPipelineRunTaskStatusProps) => {
   return (
-    <PipelineBars key={pipelineRun.metadata?.name} pipelinerun={pipelineRun} />
+    <PipelineBars key={pipelineRun?.metadata?.name} pipelinerun={pipelineRun} />
```
Comment (invincibleJai) [inline plugins/tekton/src/components/PipelineRunList/LinkedPipelineRunTaskStatus.tsx:14]: if `pipelineRun` is not there then good to handle it as an early return

**Comment by invincibleJai** [comment]
on load, I needed to click thrice for the sort to work. I tried in dev mode but the experience would be same I blv

tekonTablePaginatioin

**Comment by debsmita1** [comment]
> on load, I needed to click thrice for the sort to work. I tried in dev mode but the experience would be same I blv
> 
> tekonTablePaginatioin

Fixed it ! PTAL

File: plugins/tekton/src/components/PipelineRunList/LinkedPipelineRunTaskStatus.tsx [hunk 3]
```diff
@@ -10,9 +10,9 @@ export interface LinkedPipelineRunTaskStatusProps {
 const LinkedPipelineRunTaskStatus = ({
   pipelineRun,
 }: LinkedPipelineRunTaskStatusProps) => {
-  return (
-    <PipelineBars key={pipelineRun.metadata?.name} pipelinerun={pipelineRun} />
-  );
+  return pipelineRun?.metadata?.name ? (
+    <PipelineBars key={pipelineRun?.metadata?.name} pipelinerun={pipelineRun} />
```
Comment (invincibleJai) [inline plugins/tekton/src/components/PipelineRunList/LinkedPipelineRunTaskStatus.tsx:14]: nit:
```suggestion
    <PipelineBars key={pipelineRun.metadata.name} pipelinerun={pipelineRun} />
```

File: plugins/tekton/src/components/PipelineRunList/PipelineRunHeader.tsx [hunk 4]
```diff
@@ -1,40 +1,73 @@
...
+    defaultSort: 'desc',
+    render: (row): React.ReactNode => {
+      const startTime =
+        (row as PipelineRunKind)?.status?.startTime &&
+        new Date((row as PipelineRunKind)?.status?.startTime || '');
+      return startTime ? (
+        <Timestamp className="bs-tkn-timestamp" date={startTime} />
+      ) : (
+        '-'
+      );
```
Comment (invincibleJai) [inline plugins/tekton/src/components/PipelineRunList/PipelineRunHeader.tsx:62]:
```suggestion
      const startTime =
        (row as PipelineRunKind)?.status?.startTime || '';
      return startTime ? (
        <Timestamp className="bs-tkn-timestamp" date={startTime} />
      ) : (
        '-'
      );
```

File: plugins/tekton/src/components/common/Table.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+} from '@backstage/core-components';
+
+type TableProps<D = any> = Partial<D> & {
+  tableData: any[];
+  columns: TableColumn[];
+  title?: string;
+  subTitle?: string;
+};
+
+export const Table = ({ tableData, columns }: TableProps) => {
```
Comment (invincibleJai) [inline plugins/tekton/src/components/common/Table.tsx:15]: do we need this component now or can the caller directly use `Table` from backstage?

**Review by invincibleJai — APPROVED** [review]

**Comment by invincibleJai** [comment]
LGTM!!

**Pull request merged**